### PR TITLE
Download private ID set script

### PR DESCRIPTION
**Pull request by ChanochShayner**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
Download private ID set script.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [ ] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/Marketplace/download_private_id_set.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import os
+import json
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
+ARTIFACTS_PATH = '/home/runner/work/content-private/content-private/content/artifacts'
+PRIVATE_ID_SET_FILE = 'private_id_set.json'
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:9]: let's turn these into arguments

File: dev-requirements-py3.txt [hunk 2]
```diff
@@ -1,7 +1,7 @@
 flake8==3.8.3
 bandit==1.6.2
 demisto-py==2.0.19
-demisto-sdk==1.2.11
+git+https://github.com/demisto/demisto-sdk.git@validation_incident_fields_layout_and_mappers_exist
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply dev-requirements-py3.txt:4]: Ok

File: Tests/Marketplace/download_private_id_set.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+    is_private_id_set_file_exist = is_file_exist(public_storage_bucket, storage_client)
+
+    if is_private_id_set_file_exist:
+        index_blob = public_storage_bucket.blob(STORAGE_ID_SET_PATH)
+        index_blob.download_to_filename(f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}')
+
+    else:
+        with open(f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}', 'w') as empty_id_set:
+            json.dump(EMPTY_ID_SET, empty_id_set)
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:57]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    "IndicatorFields": [],
+    "IndicatorTypes": [],
+    "Layouts": [],
+    "Reports": [],
+    "Widgets": [],
+    "Mappers": []
+}
+
+
+def is_file_exist(public_storage_bucket, storage_client):
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:30]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    blob = public_storage_bucket.blob(STORAGE_ID_SET_PATH)
+    return blob.exists(storage_client)
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:32]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
+ARTIFACTS_PATH = '/home/runner/work/content-private/content-private/content/artifacts'
+PRIVATE_ID_SET_FILE = 'private_id_set.json'
+
+
+EMPTY_ID_SET = {
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:12]: There is no such option in sdk

File: Tests/Marketplace/download_private_id_set.py [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+    if is_private_id_set_file_exist:
+        index_blob = public_storage_bucket.blob(STORAGE_ID_SET_PATH)
+        index_blob.download_to_filename(f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}')
+
+    else:
+        with open(f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}', 'w') as empty_id_set:
+            json.dump(EMPTY_ID_SET, empty_id_set)
+
+    if os.path.exists(f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}'):
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:59]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+        return f'{ARTIFACTS_PATH}/{PRIVATE_ID_SET_FILE}'
+
+    return ''
```
[the remark this answers is not in the document]
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py:62]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
+ARTIFACTS_PATH = '/home/runner/work/content-private/content-private/content/artifacts'
+PRIVATE_ID_SET_FILE_IN_ARTIFACTS = ARTIFACTS_PATH + 'private_id_set.json'
+
+
+def upload_empty_id_set_to_bucket():
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:12]: it does not upload it to a bucket

File: Tests/Marketplace/download_private_id_set.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
+import os
+import json
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
+ARTIFACTS_PATH = '/home/runner/work/content-private/content-private/content/artifacts'
+PRIVATE_ID_SET_FILE_IN_ARTIFACTS = ARTIFACTS_PATH + 'private_id_set.json'
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:9]: these should be arguments

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/Marketplace/download_private_id_set.py (hunk 9 above)
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py]: done

File: Tests/Marketplace/download_private_id_set.py (hunk 10 above)
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+        "Layouts": [],
+        "Reports": [],
+        "Widgets": [],
+        "Mappers": []
+    }
+    with open(private_id_set_artifacts_path, 'w') as id_set:
+        json.dump(empty_id_set, id_set)
+
+
+def file_exists_in_bucket(public_storage_bucket):
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:32]: what file? let the name of the function reflect that

File: Tests/Marketplace/download_private_id_set.py [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+        public_storage_bucket (google.cloud.storage.bucket.Bucket): google storage bucket where private_id_set.json
+        is stored.
+    Returns:
+        str: private ID set file full path.
+    """
+
+    if not os.path.exists(ARTIFACTS_PATH):
+        os.mkdir(ARTIFACTS_PATH)
+
+    private_id_set_artifacts_path = ARTIFACTS_PATH + '/private_id_set.json'
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:50]:
```suggestion
    private_id_set_path = ARTIFACTS_PATH + '/private_id_set.json'
```

File: Tests/Marketplace/download_private_id_set.py [hunk 13]
```diff
@@ -0,0 +1,98 @@
+import os
+import json
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
+ARTIFACTS_PATH = '/home/runner/work/content-private/content-private/content/artifacts'
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:8]: These should be arguments

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Tests/Marketplace/download_private_id_set.py (hunk 11 above)
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py]: done

File: Tests/Marketplace/download_private_id_set.py (hunk 12 above)
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py]: done

File: Tests/Marketplace/download_private_id_set.py (hunk 13 above)
Comment (ChanochShayner) [reply Tests/Marketplace/download_private_id_set.py]: done

File: Tests/Marketplace/download_private_id_set.py [hunk 14]
```diff
@@ -0,0 +1,98 @@
+import os
+import json
+import argparse
+from Tests.Marketplace.marketplace_services import init_storage_client
+
+
+STORAGE_ID_SET_PATH = 'content/private_id_set.json'
```
Comment (guykeller) [inline Tests/Marketplace/download_private_id_set.py:7]: These all need to be args

**Review by guykeller — APPROVED** [review]
fix that last comment and we're good

**Pull request merged**